pg-ephemeral/integrations/ruby/spec/integration/pg_ephemeral_spec.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require 'pg_ephemeral'
+
+RSpec.describe PgEphemeral do
+  describe '.version' do
+    it 'returns the expected pg-ephemeral version' do
+      expected_version = ENV.fetch('EXPECTED_PG_EPHEMERAL_VERSION')
+
+      expect(PgEphemeral.version).to eq(expected_version)
+    end
+
+    it 'returns a non-empty string' do
+      expect(PgEphemeral.version).not_to be_empty
+    end
+
+    it 'returns a semantic version format' do
+      expect(PgEphemeral.version).to match(/\A\d+\.\d+\.\d+\z/)
+    end
+
+    it 'raises an error when binary execution fails' do
+      status = instance_double(Process::Status, success?: false)
+      allow(Open3).to receive(:capture3).and_return(['', '', status])
+
+      expect { PgEphemeral.version }.to raise_error(RuntimeError, 'Failed to get version from pg-ephemeral binary')
+    end
+  end
+
+  describe '.with_server' do
+    it 'yields a server with url' do
+      PgEphemeral.with_server do |server|
+        expect(server).to be_a(PgEphemeral::Server)
+        expect(server.url).to be_a(String)
+        expect(server.url).to match(/\Apostgres:\/\//)
+      end
+    end
+
+    it 'returns the block result' do
+      result = PgEphemeral.with_server { |_server| 42 }
+
+      expect(result).to eq(42)
+    end
+
+    it 'supports custom instance names' do
+      PgEphemeral.with_server(instance_name: 'custom') do |server|
+        expect(server.url).to be_a(String)
+      end
+    end
+
+    it 'allows connecting to the database and executing queries' do
+      PgEphemeral.with_server do |server|
+        connection = PG.connect(server.url)
+        result = connection.exec('SELECT 1 AS value')
+
+        expect(result.first['value']).to eq('1')
+
+        connection.close
```

Comment:
i guess it doesn't really matter that this line won't run when the assertion fails because the server will be torn down anyway. just triggered my spidey sense that this isn't in an ensure.